backend/src/test/java/com/bakdata/conquery/integration/tests/ImportUpdateTest.java

```diff
@@ -0,0 +1,127 @@
+
+		final ImportId importId = ImportId.Parser.INSTANCE.parse(dataset.getName(), "test_table2", "test_table2");
+		final ImportId importId2 = ImportId.Parser.INSTANCE.parse(dataset.getName(), "test_table", "test_table");
+		final ImportId importId3 = ImportId.Parser.INSTANCE.parse(dataset.getName(), "test_table3", "test_table3");
+
+		QueryTest test = (QueryTest) JsonIntegrationTest.readJson(dataset, testJson);
+		QueryTest test2 = (QueryTest) JsonIntegrationTest.readJson(dataset, testJson2);
+
+		List<File> preprocessedFiles;
+
+		// Manually import data, so we can do our own work.
+		{
+			ValidatorHelper.failOnError(log, conquery.getValidator().validate(test));
+
+			importSecondaryIds(conquery, test.getContent().getSecondaryIds());
+			conquery.waitUntilWorkDone();
+
+			LoadingUtil.importTables(conquery, test.getContent().getTables());
+			conquery.waitUntilWorkDone();
+
+			LoadingUtil.importConcepts(conquery, test.getRawConcepts());
+			conquery.waitUntilWorkDone();
+
+			LoadingUtil.importTableContents(conquery, test.getContent().getTables(), conquery.getDataset());
+			conquery.waitUntilWorkDone();
+		}
+
+
+		// State before update.
+		{
+			log.info("Checking state before update");
+
+			// Must contain the import.
+			assertThat(namespace.getStorage().getAllImports())
+					.filteredOn(imp -> imp.getId().equals(importId))
+					.isNotEmpty();
+
+			assertThat(namespace.getStorage().getCentralRegistry().getOptional(importId))
+					.isNotEmpty();
+
+			for (ShardNode node : conquery.getShardNodes()) {
+				for (Worker worker : node.getWorkers().getWorkers().values()) {
+					if (!worker.getInfo().getDataset().equals(dataset.getId())) {
+						continue;
+					}
+
+					final ModificationShieldedWorkerStorage workerStorage = worker.getStorage();
+
+					assertThat(workerStorage.getAllCBlocks())
+							.describedAs("CBlocks for Worker %s", worker.getInfo().getId())
+							.filteredOn(block -> block.getBucket().getId().getDataset().equals(dataset.getId()))
+							.isNotEmpty();
+					assertThat(workerStorage.getAllBuckets())
+							.filteredOn(bucket -> bucket.getId().getDataset().equals(dataset.getId()))
+							.describedAs("Buckets for Worker %s", worker.getInfo().getId())
+							.isNotEmpty();
+
+					// Must contain the import.
+					assertThat(workerStorage.getImport(importId))
+							.isNotNull();
+				}
+			}
+
+		}
+
+		//Update import
+		{
+			preprocessedFiles = LoadingUtil.generateCqpp(conquery, test2.getContent().getTables());
+			Assert.assertNotNull(preprocessedFiles);
+			Assert.assertEquals(2, preprocessedFiles.size());
+
+			//different ImportIds
+			LoadingUtil.updateCqppFile(conquery, preprocessedFiles.get(0), importId2, Response.Status.Family.CLIENT_ERROR, "Conflict");
+			conquery.waitUntilWorkDone();
+
+			//same ImportIds but Import is not present
+			LoadingUtil.updateCqppFile(conquery, preprocessedFiles.get(1), importId3, Response.Status.Family.CLIENT_ERROR, "Not Found");
+			conquery.waitUntilWorkDone();
```

Comment:
In diesem Fall ist doch nicht nur der Import nicht da, sondern die komplette Tabelle fehlt